common/src/jni/main/cpp/conscrypt/native_crypto.cc

```diff
@@ -3775,7 +3776,148 @@ static jint NativeCrypto_EVP_AEAD_CTX_open_buf(JNIEnv* env, jclass, jlong evpAea
                            inBuffer, aadArray, EVP_AEAD_CTX_open);
 }
 
+static jlong NativeCrypto_CMAC_CTX_new(JNIEnv* env, jclass) {
+    CHECK_ERROR_QUEUE_ON_RETURN;
+    JNI_TRACE("CMAC_CTX_new");
+    auto cmacCtx = CMAC_CTX_new();
+    if (cmacCtx == nullptr) {
+        conscrypt::jniutil::throwOutOfMemory(env, "Unable to allocate CMAC_CTX");
+        return 0;
+    }
+
+    return reinterpret_cast<jlong>(cmacCtx);
+}
+
+static void NativeCrypto_CMAC_CTX_free(JNIEnv* env, jclass, jlong cmacCtxRef) {
+    CHECK_ERROR_QUEUE_ON_RETURN;
+    CMAC_CTX* cmacCtx = reinterpret_cast<CMAC_CTX*>(cmacCtxRef);
+    JNI_TRACE("CMAC_CTX_free(%p)", cmacCtx);
+    if (cmacCtx == nullptr) {
+        return;
+    }
+    CMAC_CTX_free(cmacCtx);
```

Comment:
BoringSSL's free functions all tolerate null, so no need for the null check.

Reply:
Thanks! These (this and `HMAC_free`) should actually be throwing on null which I missed.  The create path throws on null, so the only way a null can get here is a Conscrypt bug.  And it feels a bit weird to call the free function if we're throwing anyway.

I'm slowly working towards more consistent enforcement of the invariant that passing a null pointer from java into jni is always an error.